src/consensus/aft/raft.h

```diff
@@ -1890,23 +1905,20 @@ namespace aft
       // receiving a conflicting AppendEntries
       rollback(last_committable_index());
 
-      if (can_endorse_primary())
-      {
-        state->leadership_state = kv::LeadershipState::Follower;
-        RAFT_INFO_FMT(
-          "Becoming follower {}: {}.{}",
-          state->node_id,
-          state->current_view,
-          state->commit_idx);
+      state->leadership_state = kv::LeadershipState::Follower;
```

Comment:
I am not sure why we had this condition, but it would prevent a node from reverting to follower after retirement, which is clearly a problem when they need to stay on longer to help commit the new term that follows their retirement.